File: 485.max-consecutive-ones/solution.kt
class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var currentLen = 0
        var maxLen = 0
        for (num in nums) {
            currentLen = when (num) {
                1 -> currentLen + 1
                else -> 0
            }
            if (currentLen > maxLen)
                maxLen = currentLen
        }
        return maxLen
    }
}

File: 485.max-consecutive-ones/solution.go
func findMaxConsecutiveOnes(nums []int) int {
    max, run := 0, 0
    for _, num := range nums {
        if num == 1 {
            run++
            if run > max {
                max = run
            }
        } else {
            run = 0
        }
    }
    return max
}